src/cli/llm/error.rs

```diff
@@ -1,11 +1,46 @@
-use derive_more::From;
-use strum_macros::Display;
+use reqwest::StatusCode;
+use thiserror::Error;
 
-#[derive(Debug, From, Display, thiserror::Error)]
+#[derive(Debug, Error)]
+pub enum WebcError {
+    #[error("Response failed with status {status}: {body}")]
+    ResponseFailedStatus { status: StatusCode, body: String },
+    #[error("Reqwest error: {0}")]
+    Reqwest(#[from] reqwest::Error),
+}
+
+#[derive(Debug, Error)]
 pub enum Error {
+    #[error("GenAI error: {0}")]
     GenAI(genai::Error),
+    #[error("Webc error: {0}")]
+    Webc(WebcError),
+    #[error("Empty response")]
     EmptyResponse,
-    Serde(serde_json::Error),
+    #[error("Serde error: {0}")]
+    Serde(#[from] serde_json::Error),
+}
+
+impl From<genai::Error> for Error {
+    fn from(err: genai::Error) -> Self {
```

Comment:
You might not need so much code since this is merged — https://github.com/jeremychone/rust-genai/commit/736bbec943abe1d3f7d0dbc7106e420240458c5b

Reply:
Oh, great.

Reply:
@tusharmath Pls, why is this repo: https://github.dev/laststylebender14/rust-genai being used instead of the main genai repo: https://github.com/jeremychone/rust-genai


jeremychone's latest lib.rs:

```rs
// region:    --- Modules

mod support;

mod client;
mod common;
mod error;

// -- Flatten
pub use client::*;
pub use common::*;
pub use error::{Error, Result};

// -- Public Modules
pub mod adapter;
pub mod chat;
pub mod resolver;
pub mod webc;

// endregion: --- Modules
```
laststylebender14's latest commit hash lib.rs:

```rs
// region:    --- Modules

mod support;
mod webc;

mod client;
mod common;
mod error;

// -- Flatten
pub use client::*;
pub use common::*;
pub use error::{Error, Result};

// -- Public Modules
pub mod adapter;
pub mod chat;
pub mod resolver;

// endregion: --- Modules
```


So the webc is still private.

Reply:
@laststylebender14 can you update your Repo with the latest changes? Also can you transfer ownership to `tailcallhq`

Reply:
@onyedikachi-david I have taken the latest changes here — https://github.com/tailcallhq/rust-genai

Reply:
Okay

Reply:
please apply discussed changes:
- use https://github.com/tailcallhq/rust-genai as crate
- use automatic conversion from webc::Error